src/cpp/handler/handlerengine.cpp

```diff
@@ -1251,7 +1249,14 @@ class HandlerEngine::Private : public QObject
 	Connection controlReqReadyConnection;
 	Connection controlServerConnection;
 	Connection itemReadyConnection;
-	map<Deferred*, Connection> finishedConnection;
+	map<Deferred*, Connection> finishedDConnection;
+	map<RefreshWorker*, Connection> finishedRWConnection;
+	map<AcceptWorker*, Connection> finishedAWConnection;
+	map<InspectWorker*, Connection> finishedIWConnection;
+	map<ConnCheckWorker*, Connection> finishedCCWConnection;
+	map<Subscription*, Connection> subscribedConnection;
+	map<AcceptWorker*, Connection> retryPacketReadyConnection;
+	map<AcceptWorker*, Connection> sessionsReadyConnection;
```

Comment:
RefreshWorker and ConnCheckWorker both inherit Deferred, so I think they could stay using the `finishedConnection` map?

For AcceptWorker, this could be an opportunity to try a map of a list.

Reply:
ah thats why did work in the first place! 
all thoes workers including `AcceptWorker ` inherit `Deferred` . all replaced